wled00/FX_fcn.cpp

```diff
@@ -470,6 +557,7 @@ void Segment::setMode(uint8_t fx, bool loadDefaults) {
         sOpt = extractModeDefaults(fx, "mY");   if (sOpt >= 0) mirror_y  = (bool)sOpt; // NOTE: setting this option is a risky business
         sOpt = extractModeDefaults(fx, "pal");  if (sOpt >= 0) setPalette(sOpt); //else setPalette(0);
       }
+      markForReset();
```

Comment:
This is a good change,  resetting mid-transition like in `main` led to quite a few issues (new effect allocating data just for it to reset and re-allocate mid-transition), so I will probably change it there too for b4. Should probably be un-commented in `WS2812FX::setMode` as well.